packages/flutter/lib/src/widgets/async.dart

```diff
@@ -0,0 +1,256 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
```

Comment:
nits: some things that the style guide goes into more detail that i happen to notice - constructor should go first, onConnecting et al should be renamed according to the naming scheme described in the style guide, please don't use abbreviations in identifier names (e.g. "o"), please don't use `=>` in multiline expressions; if you have a newline after an open parenthesis, please have a newline before the matching close parenthesis. Also, in general, please prefer readability over strict adherence to 80-character limits.